feat: add new nodejs Azure trigger templates

mydiemho opened the pull request:
<!--
1. Please check out and follow our Contributing Guidelines: https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md
2. Do not remove any section of the template. If something is not applicable leave it empty but leave it in the PR
3. Please follow the template, otherwise we'll have to ask you to update it and it will take longer until your PR is merged
-->

## What did you implement:

<!--
Briefly describe the feature if no issue exists for this PR
-->

Add nodejs templates for the following Azure Functions Trigger:
- [x] Storage Blob
- [x] Storage Queue
- [x] Service Bus Queue
- [x] Service Bus Topic
- [x] EventHub
- [x] Timer

Update existing Http trigger

## How did you implement it:

<!--
If this is a nontrivial change please briefly describe your implementation so its easy for us to understand and review your code.
-->

## How can we verify it:

<!--
Add any applicable config, commands, screenshots or other resources
to make it easy for us to verify this works. The easier you make it for us
to review a PR, the faster we can review and merge it.

Examples:
* serverless.yml - Fully functioning to easily deploy changes
* Screenshots - Showing the difference between your output and the master
* Cloud Configuration - List cloud resources and show that the correct configuration is in place (e.g. AWS CLI commands)
* Other - Anything else that comes to mind to help us evaluate
-->

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [ x] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [ x] Write documentation
- [ x] Ensure there are no lint errors.  
       **Validate via `npm run lint-updated`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [ x] Ensure introduced changes match Prettier formatting.  
       **Validate via `npm run prettier-check-updated`**  
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [x ] Make sure code coverage hasn't dropped
- [x ] Provide verification config / commands / resources
- [x ] Enable "Allow edits from maintainers" for this PR
- [x ] Update the messages below

**_Is this ready for review?:_** YES  
**_Is it a breaking change?:_** NO

tbarlow12 commented on lib/plugins/create/templates/azure-nodejs-service-bus-queue/src/handlers/hello.js:
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+module.exports.sayHello = async function(context, req) {
+  context.log('JavaScript HTTP trigger function processed a request.');
```
Does it make sense to include http handlers in the template for `service-bus-queue` (or any other type besides HTTP for that matter? I understand that people could use it this way, but it might be a bit confusing to see in a template

mydiemho replied:
the http trigger is just there to demonstrate how to have 2 function in the yaml.  Not sure if this what you're asking

pmuens commented on lib/plugins/create/templates/azure-nodejs-event-hub/package.json:
```diff
@@ -0,0 +1,19 @@
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": "^1.0.0-26"
```
Should we just go with `"*"` as the version number here? We faced some issues in the past where users used the Google templated and got errors because the template was not updated.

Wondering if `"*"` might make sense here since you always want the most recent version when you create a template. Otherwise, if `"*"` is too wild we might want to pin it to the major version number.

@medikoo any experiences with that?

medikoo replied:
I believe we should stick here to major version for which template was created (so `^1.0.0` in this case). If there's a major bump on a plugin, there's a risk that template will no longer work with it, so guess it's a good practice to revisit template and update it after each major bump.

I would definitely not refer to pre-release here (afaik `^` is not effective for pre-release versions.. so above range simply equals static version of`1.0.0-26`). Let's wait for official `1.0.0` release.

pmuens commented on lib/plugins/create/templates/azure-nodejs-service-bus-queue/package.json:
```diff
@@ -0,0 +1,19 @@
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": "^1.0.0-26"
```
Same comment applies here

pmuens commented on lib/plugins/create/templates/azure-nodejs-service-bus-topic/package.json:
```diff
@@ -0,0 +1,19 @@
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": "^1.0.0-26"
```
Same comment applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs-storage-blob/README.md:
```diff
@@ -0,0 +1,133 @@
+# Azure Functions - Quick Start
```
Same comment applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs-storage-blob/package.json:
```diff
@@ -0,0 +1,19 @@
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": "^1.0.0-26"
```
Same comment applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs-storage-queue/README.md:
```diff
@@ -0,0 +1,133 @@
+# Azure Functions - Quick Start
```
Same comment applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs-storage-queue/package.json:
```diff
@@ -0,0 +1,19 @@
+  "dependencies": {},
+  "devDependencies": {
+    "azurite": "^2.7.0",
+    "serverless-azure-functions": "^1.0.0-24"
```
Same comment applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs-timer/README.md:
```diff
@@ -0,0 +1,133 @@
+# Azure Functions - Quick Start
```
Same comment applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs-timer/package.json:
```diff
@@ -0,0 +1,17 @@
+  ],
+  "dependencies": {},
+  "devDependencies": {
+    "serverless-azure-functions": "^1.0.0-24"
```
Same commet applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs/package.json:
```diff
@@ -12,6 +12,6 @@
   "dependencies": {},
   "devDependencies": {
-    "serverless-azure-functions": ">=0.7.0"
+    "serverless-azure-functions": "^1.0.0-26"
```
Same comment applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs-event-hub/README.md:
```diff
@@ -0,0 +1,133 @@
+# Azure Functions - Quick Start
```
This file seems to be copied over from the docs and might need some overhaul in terms of wording and links to the actual docs.

pmuens commented on lib/plugins/create/templates/azure-nodejs-service-bus-queue/README.md:
```diff
@@ -0,0 +1,133 @@
+# Azure Functions - Quick Start
```
Same comment applies here.

pmuens commented on lib/plugins/create/templates/azure-nodejs-service-bus-topic/README.md:
```diff
@@ -0,0 +1,133 @@
+# Azure Functions - Quick Start
```
Same comment applies here.

mydiemho commented:
> Thanks a lot for working on this @mydiemho 👍
> 
> I just looked through the PR and added some comments. While reading through it I thought about the same @medikoo brought up in his PR review. Can we somehow compress all the 6 templates into 1 template for `azure-nodejs` (we for example have commented out sections for the different AWS event sources in the `aws-*` templates) and move the specifics into the `examples` repo and cross reference from the templates readme to the examples?
> 
> Overall thanks a lot for the hard work on the plugin and the docs / templates! 👌

@pmuens  @medikoo addressed your comments and going to create new prs for updating existing nodejs template and adding new samples

Here's the pr for updating azure-nodejs #6626
